modules/samples/firealarm/component/manager/src/main/java/org/homeautomation/firealarm/manager/api/ManagerService.java

```diff
@@ -0,0 +1,337 @@
+                            @QueryParam("name") String name) {
+        DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+        deviceIdentifier.setId(deviceId);
+        deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+        try {
+            if (APIUtil.getDeviceManagementService().isEnrolled(deviceIdentifier)) {
+                response.setStatus(Response.Status.CONFLICT.getStatusCode());
+                return false;
+            }
+
+            String owner = APIUtil.getAuthenticatedUser();
+            Device device = new Device();
+            device.setDeviceIdentifier(deviceId);
+            EnrolmentInfo enrolmentInfo = new EnrolmentInfo();
+            enrolmentInfo.setDateOfEnrolment(new Date().getTime());
+            enrolmentInfo.setDateOfLastUpdate(new Date().getTime());
+            enrolmentInfo.setStatus(EnrolmentInfo.Status.ACTIVE);
+            enrolmentInfo.setOwnership(EnrolmentInfo.OwnerShip.BYOD);
+            device.setName(name);
+            device.setType(DeviceTypeConstants.DEVICE_TYPE);
+            enrolmentInfo.setOwner(owner);
+            device.setEnrolmentInfo(enrolmentInfo);
+            KeyGenerationUtil.createApplicationKeys(DeviceTypeConstants.DEVICE_TYPE);
+            TokenClient accessTokenClient = new TokenClient(DeviceTypeConstants.DEVICE_TYPE);
+            AccessTokenInfo accessTokenInfo = accessTokenClient.getAccessToken(owner, deviceId);
+
+            //create token
+            String accessToken = accessTokenInfo.getAccess_token();
+            String refreshToken = accessTokenInfo.getRefresh_token();
+            List<Device.Property> properties = new ArrayList<>();
+            Device.Property accessTokenProperty = new Device.Property();
+            accessTokenProperty.setName("accessToken");
+            accessTokenProperty.setValue(accessToken);
+            Device.Property refreshTokenProperty = new Device.Property();
+            refreshTokenProperty.setName("refreshToken");
+            refreshTokenProperty.setValue(refreshToken);
+            properties.add(accessTokenProperty);
+            properties.add(refreshTokenProperty);
+            device.setProperties(properties);
+
+            boolean added = APIUtil.getDeviceManagementService().enrollDevice(device);
+            if (added) {
+                response.setStatus(Response.Status.OK.getStatusCode());
+            } else {
+                response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+            }
+            return added;
+        } catch (DeviceManagementException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            log.error(e.getErrorMessage(), e);
+            return false;
+        } catch (AccessTokenException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            log.error("Unable to obtain access token", e);
+            return false;
```

Comment:
The standard now is to return a Response object with a status code set and a payload attached if any.
Change the code accordingly.